app/models/theme_setting.rb

```diff
@@ -0,0 +1,64 @@
+class ThemeSetting < ActiveRecord::Base
+  belongs_to :theme
+
+  validates_presence_of :name, :theme
+  validates :data_type, numericality: { only_integer: true }
+  validates :name, length: { maximum: 255 }
+
+  after_commit do
```

Comment:
I wonder if this should be after_save for extra simplicity, cause if you roll back you may corrupt cache?